### ISSUE TITLE
ZeroOneLoss; Margin-based or not

### ISSUE THREAD

**Issue by Evizero** [JuliaML/Losses.jl#49]
I remember the question being raised why the `ZeroOneLoss` is not a subtype of `MarginLoss` (although I can't find the associated issue).

I investigated a bit to retrace my steps. The book I have been using as a guide states the classification Loss is not margin-based, yet our implementation seems to fit the definition. I found out that the reason for this is that I simply did not implement the loss as it is described in the book (which is L(y,t) = 1 (y sign t), where the 1 is an indicator function). The book version clearly isn't margin based, since it can't be written as L(y*t).

At some point I must have been compelled to try and formulate the Loss as a margin-based one, but then forgot to reflect that in the type tree.

tl;dr: Our implementation is margin-based. will fix.
